main.py

```diff
@@ -38,77 +41,84 @@ def login(self):
         # print("The login get cookies are: ")
         # print(login_get.cookies)
 
-    def get_req(self, object):
-        #get the object for ULR
+    def get_req(self, object, offset=None, limit=None):
+
         g1 = requests.get('https://wger.de/api/v2/')
-        g = requests.get(url=g1.json()[object], headers=self.header)
-        if g.status_code == 200:
-            return g.json()
+        url = g1.json().get(object)
+        if url != None:
+            if offset is None and limit is None:
+                url1 = url
+            else:
+                url1 = url + f"?limit={limit}&offset={offset}"
+
+            g = requests.get(url=url1, headers=self.header)
+            if g.status_code == 200:
+                return g.json()
+            else:
+                return g.status_code, "Couldn't make the request!"
         else:
-            return "Couldn't make the request"
+            return False, 'Invalid url'
+
 
     def post_req(self,object, data):
 
         g1 = requests.get('https://wger.de/api/v2/')
         url1 = g1.json().get(object)
+
         if url1 != None:
             req = requests.post(url=url1,data=data,headers=self.header)
             if req.status_code == 201:
-                return f'The post request for {object} was done succusefuly'
+                return req, f'The post request for {object} was done succusefuly'
             else:
-                return f'Error {req.status_code}'
+                return {req.status_code}, 'Error  {req.status_code}'
         else:
-            return 'Invalid url'
-
+            return False, 'Invalid url'
 
 
     def delete_req(self,object, id):
         g1 = requests.get('https://wger.de/api/v2/')
         url1 = g1.json().get(object)
-        url1 = f"{url1}/{id}"
         if url1 != None:
-            req = requests.delete(url=url1, headers=self.header)
+            url = f"{url1}/{id}"
+            req = requests.delete(url=url, headers=self.header)
+            if req.status_code not in (200,202,204):
+                return req.status_code, f"The delete for id {id} couldn't be performed"
             return req
```

Comment:
Here you should also return a tuple of 'req' and a message reflecting the state of the method.